src/main/java/com/bhoruka/bloodbank/model/request/SearchCampRequest.java

```diff
@@ -0,0 +1,25 @@
+package com.bhoruka.bloodbank.model.request;
+
+import java.util.Date;
+
+import lombok.Builder;
+import lombok.Value;
+
+@Value
+@Builder
+public class SearchCampRequest {
+
+    private String partnerId;
+
+    private Date dateOfCamp;
+
+    private String expectedNoOfDonors;
+
+    private String actualNoOfDonors;
+
+    Integer pageNo;
```

Comment:
add validation to ensure this is greater than 0.

Reply:
forgot to mention, you can do this very cleanly by just using jsrValidations. Add hibernate-validator as a dependency.